hub/codec/codecs.py

```diff
@@ -0,0 +1,380 @@
+from io import BytesIO
+import numcodecs
+from numcodecs.abc import Codec
+
+import numpy as np
+from PIL import Image
+
+
+class Base:
+    """Base class for numcodec compressors"""
+
+    def encode(self, array: np.ndarray) -> bytes:
+        raise NotImplementedError()
+
+    def decode(self, bytes: bytes) -> np.ndarray:
+        raise NotImplementedError()
+
+
+class NumPy(Base):
+    """Numpy compressor"""
+
+    def __init__(self):
+        super().__init__()
+
+    def encode(self, array: np.ndarray) -> bytes:
+        """
+        Encode given array
+
+        Example:
+            arr = np.arange(100, 100, 2, dtype='uint8')
+            arr_encoded = numpy_codec.encode(x)
+
+        Args:
+            array (np.ndarray): Data to be encoded
+
+        Returns:
+            Encoded data.
+        """
+        with BytesIO() as f:
+            np.save(f, array, allow_pickle=True)
+            return f.getvalue()
+
+    def decode(self, bytes_: bytes) -> np.ndarray:
+        """
+        Decode data from buffer.
+
+        Example:
+            arr_decoded = numpy_codec.decode(arr_encoded)
+
+        Args:
+            bytes_ (bytes): Encoded data
+
+        Returns:
+            Decoded data.
+        """
+        with BytesIO(bytes_) as f:
+            return np.load(f, allow_pickle=True)
+
+
+class Lz4(Base):
+    """Lz4 compressor"""
+
+    def __init__(self, acceleration: int):
+        """
+        Initialize Lz4 compressor
+
+        Args:
+            acceleration (int): Acceleration level.
+            The larger the acceleration value, the faster the algorithm, but also the lesser the compression.
```

Comment:
I don't really have a sense of scale for the acceleration. Can you provide some examples for what "faster" means?

For example, is it common for acceleration to be `> 10`? A link to documentation would also help

Reply:
Checked previously the acceleration differences though on a small subset of coco:
acceleration 1 - 93 s, 145 MB
acceleration 10 - 100 s, 155 MB
acceleration 22 - 80 s, 160 MB
Numcodecs has really bad, almost no documentation here: https://numcodecs.readthedocs.io/en/stable/

Reply:
Checked previously the acceleration differences though on a small subset of coco:
acceleration 1 - 93 s, 145 MB
acceleration 10 - 100 s, 155 MB
acceleration 20 - 80 s, 160 MB
Numcodecs has really bad, almost no documentation here: https://numcodecs.readthedocs.io/en/stable/

Reply:
http://fastcompression.blogspot.com/2015/04/sampling-or-faster-lz4.html
Found a better reference. Might be nice to link _something_ in the docstring so interested parties know where to start